Gemfile

```diff
@@ -52,6 +52,13 @@ gem 'sneakers'
 gem 'sprockets', '>= 3.7.2'
 gem 'turbolinks'
 
+## Added for Ruby 3.1 support
+gem "matrix"
+gem "net-imap", require: false
+gem "net-pop", require: false
+gem "net-smtp", require: false
```

Comment:
Question: did you find you needed all of these, or did you pull these from the work on another project? I saw them but the only one I needed to get tests passing for dpul was net-smtp so I didn't add the others. I'm wondering if I should.

Reply:
Oh I just copied them. Hm, I can try removing them - we don't send email..